Portal/vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/dao/mysql/PlatformConnection.java

```diff
@@ -59,154 +59,164 @@
     private final static Logger logger = Logger.getLogger(PlatformConnection.class);
     private static PlatformConnection instance;
     private boolean firstExecution;
-    private Connection connection;
-
-    public synchronized static PlatformConnection getInstance() throws DAOException {
+    private DataSource dataSource;
 
+    public synchronized static PlatformConnection getInstance() {
         if (instance == null) {
             instance = new PlatformConnection();
         }
         return instance;
     }
 
-    private PlatformConnection() throws DAOException {
-
+    private PlatformConnection() {
         firstExecution = true;
         connect();
         createTables();
     }
 
     private void connect() {
```

Comment:
The connection is no longer retrieved in this method. Maybe we could rename it accordingly?

Reply:
What do you suggest to rename: the class name PlatformConnection, or the method name of connect() ?

PlatformConnection is still the singleton allowing the server to get a database connection.
connect() could be renamed to initDatasource().

Reply:
The method name. I find your suggestion, initDatasource(), very good, go for it.